### PR TITLE
Support log_module in router metadata

### PR DESCRIPTION
**Pull request by RudolfMan**
Initially was working on https://github.com/phoenixframework/phoenix/issues/4721, but turns out there is already an issue related https://github.com/phoenixframework/phoenix/issues/4686

### CODE REVIEW

File: lib/phoenix/router/scope.ex [hunk 1]
```diff
@@ -40,6 +40,7 @@ defmodule Phoenix.Router.Scope do
       opts
       |> Keyword.get(:metadata, %{})
       |> Map.put(:log, Keyword.get(opts, :log, top.log))
+      |> Map.put_new(:log_module, plug)
```
Comment (josevalim) [inline lib/phoenix/router/scope.ex:43]: I don't think we should put this by default, so we generate less metadata. Then in the formatter we do this:

    metadata[:log_module] || plug

WDYT?

**Comment by josevalim** [comment]
Thank you for the PR! I dropped a comment. Note we also need to change here: https://github.com/phoenixframework/phoenix/blob/master/lib/phoenix/logger.ex#L282

File: lib/phoenix/router/scope.ex (hunk 1 above)
Comment (RudolfMan) [reply lib/phoenix/router/scope.ex:43]: I like the idea! 
will make the change

**Comment by josevalim** [comment]
Looks great! Also feel welcome to send a PR to LiveView that makes use of this new option. :)

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**